org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/exceptions/generateRestException.mtl

```diff
@@ -0,0 +1,92 @@
+[comment encoding = UTF-8 /]
+[comment
+/*******************************************************************************
+ * Copyright (c) 2020 Jad El-khoury and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompanies this distribution.
+ *
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ *******************************************************************************/
+/]
+
+[module generateRestException('http://org.eclipse.lyo/oslc4j/adaptorInterface')]
+
+[import org::eclipse::lyo::oslc4j::codegenerator::services::adaptorInterfaceServices/]
+
+[template public generateRestException(anAdaptorInterface : AdaptorInterface)]
+[file (javaClassFullFileNameForRestException(anAdaptorInterface), false, 'UTF-8')]
+// [protected ('Copyright')]
+/*******************************************************************************
+ * Copyright (c) 2012 IBM Corporation and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompanies this distribution.
+ *
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * This file is generated by org.eclipse.lyo.oslc4j.codegenerator
+ *******************************************************************************/
+// [/protected]
+
+package [javaClassPackageNameForRestException(anAdaptorInterface) /];
+
+import javax.servlet.http.HttpServletResponse;
+
+// [protected ('imports')]
+// [/protected]
+
+// [protected ('pre_class_code')]
+// [/protected]
+
+/* Holds a status code and error message for an error response. */
+public class [anAdaptorInterface.javaClassNameForRestException()/] extends Exception {
+    private int statusCode;
+    private String message;
+
+    // [protected ('class_attributes')]
+    // [/protected]
+
+    // [protected ('class_methods')]
+    // [/protected]
+
+    public [anAdaptorInterface.javaClassNameForRestException()/](int statusCode, String message) {
+        this.statusCode = statusCode;
+        this.message = message;
+    }
+    
+    public [anAdaptorInterface.javaClassNameForRestException()/](Throwable t) {
+        this(t, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Are you suggesting we don't  have the RESTException class at all, and use WebApplicationException istead?
What do you mean by "do the trick" otherwise.

Reply:
https://docs.oracle.com/javaee/7/api/javax/ws/rs/NotAuthorizedException.html can be caught on line https://github.com/eclipse/lyo.designer/pull/152/files/c5911297bf0b8af2fe631ea2a972fd7cd20d39b7#diff-1122daca959bf58a35f513dba8cff6e7R169